pkg/build/definition_parser_deffile.go

```diff
@@ -9,13 +9,156 @@ package build
 
 import (
 	"bufio"
+	"bytes"
+	"fmt"
 	"io"
 	"log"
 	"regexp"
 	"strings"
 	"unicode"
 )
 
+// validSections just contains a list of all the valid sections a definition file
+// could contain. If any others are found, an error will generate
+var validSections = map[string]bool{
```

Comment:
I get that the map to bool is needed for a nice lookup, given go has no 'in' operator... but it's kinda ugly. Is it any better just to have this in a function `sectionValid(section string) bool` which directly uses a switch/case block?

Reply:
you can do name check as the actual parser is doing, without a map just for boolean types

Reply:
The reason I did it in a map is because a go map is actually a hash table under the hood, so lookup is O(1).